src/Resources/npm-package/js/contao-filter-bundle.js

```diff
@@ -61,7 +61,28 @@ class FilterBundle {
                 FilterBundle.asyncSubmit(element.form, element);
             });
 
+        FilterBundle.registerFormAsyncSubmitEvents();
         FilterBundle.initAsyncSubmitOnInput();
+
+        document.addEventListener('filterAjaxComplete', (event) => {
+            FilterBundle.registerFormAsyncSubmitEvents();
+        });
+    }
+
+    /**
+     * Register events for form async submit.
+     *
+     * These events need to be reintialized after form reload.
+     */
+    static registerFormAsyncSubmitEvents()
+    {
+        document.querySelector('.mod_filter form[data-async] input[type="text"]').addEventListener('keydown', (event) => {
+            if (event.code == 'Enter') {
```

Comment:
can we be more strict here? or even use `event.which` to obtain numerical keycode?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent says event.which is deprecated. It is recommended to use event.key.

Reply:
Changed is to strict comparison with event.key. I think that's a good solution?

Reply:
Ok, then is event.key the best solution, looks good for me